From 88047688c47e49d1124075739cda9be4762e7968 Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Mon, 1 Jun 2015 13:25:05 +0800
Subject: [PATCH] MLK-11015-1 arm: rpmsg: add rpmsg for imx7d

enable the rpmsg for imx7d amp platform.
Up to now, imx7d sdb and imx7d 12x12 lpddr3 arm2 boards
are verified.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts | 4 ++++
 arch/arm/boot/dts/imx7d-sdb-m4.dts               | 4 ++++
 arch/arm/boot/dts/imx7d.dtsi                     | 5 +++++
 3 files changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
index afb17c4..d4d5d25 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-m4.dts
@@ -19,6 +19,10 @@
 	status = "okay";
 };
 
+&rpmsg{
+	status = "okay";
+};
+
 &iomuxc {
 	uart2 {
 		pinctrl_uart2_1: uart2grp-1 {
diff --git a/arch/arm/boot/dts/imx7d-sdb-m4.dts b/arch/arm/boot/dts/imx7d-sdb-m4.dts
index d1a7736..e9a92c9 100644
--- a/arch/arm/boot/dts/imx7d-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-m4.dts
@@ -35,6 +35,10 @@
 	status = "okay";
 };
 
+&rpmsg{
+	status = "okay";
+};
+
 &iomuxc {
 	uart2 {
 		pinctrl_uart2_1: uart2grp-1 {
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 18a2149..16f746f0 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -1141,6 +1141,11 @@
 				status = "disabled";
 			};
 
+			rpmsg: rpmsg{
+				compatible = "fsl,imx7d-rpmsg";
+				status = "disabled";
+			};
+
 			sema4: sema4@30ac0000 {
 				compatible = "fsl,imx7d-sema4";
 				reg = <0x30ac0000 0x10000>;
-- 
1.8.0

